### DISCUSSION TITLE
Welcome to ME3255!

### DISCUSSION THREAD

**Discussion by cooperrc** [cooperrc/computational-mechanics#12] [General]
Greetings!

Introduce yourself to our computational mechanics learning community. Answer these 3 questions

1. Who are you? What is a interesting fact about you?
2. Why are you here? How can this course help your career?
3. What would you change about the course to help you learn/achieve your career goals?

**Comment by taa18009** [comment]
My name Is Taryn Asselin, an interesting fact about me is I attend Siena College and play on the women's lacrosse team there. I'm here because I am an applied physics major on the mechanical engineering track. I aspire to be a sports engineer, computational mechanics will be a helpful course in this career as being good with computer systems and coding will be important to determine different designs of sports equipment to maximize efficiency in different areas of sports. I would really like to use coding in a manner that we would in the workforce, a lot of my past coding classes haven't really been applying what we are learning to physics but rather more statistics based.

**Reply by cooperrc** [reply]
>  I attend Siena College 
Welcome to our virtual UConn campus!

> and play on the women's lacrosse team there
:lacrosse: awesome!

> a lot of my past coding classes haven't really been applying what we are learning to physics but rather more statistics based

This is my gripe with a lot of computational resources+textbooks. They'll spend whole chapters dedicated to the benefits of [object oriented programming](https://en.wikipedia.org/wiki/Object-oriented_programming) and build functions from scratch, but then there's no engineering application. 

I'm trying to build something a little more hands-on here where you can see pieces of existing functions and build what you need to solve problems. 

Hope you like the course content!

**Comment by Aidan-Myatt** [comment]
Hello everyone,
I am Aidan, and one interesting fact about me is my family has five cats back home thanks to my mom. I am taking this class to progress towards getting my Mechanical Engineering degree and to lighten my course load for my senior year. This will help me in my career by allowing me to be better at coding to solve future engineering problems. If there was something I would change about this course, it would be to make it longer, so we have time to think about the problems.

**Reply by tempcooper** [reply]
> five cats back home thanks to my mom

Holy meow! :cat: 

> solve future engineering problems

Even _current_ engineering problems need better programs. The latest example was the [boeing 737 MAX failure](https://www.seattletimes.com/business/boeing-aerospace/failed-certification-faa-missed-safety-issues-in-the-737-max-system-implicated-in-the-lion-air-crash/)

**Comment by Aarons-Joshua** [comment]
Hello!

My name is Joshua, and I am going into my senior year and graduating in 2023. One interesting fact about me is that I train in the sport of Parkour (yes that means I can do flips and tricks). I am here to learn more about computational mechanics as not only is it required for my degree, but I also would love to learn more about programming in python and using the language efficiently. If I could change one thing about this course, it would be the time I am taking it, as the summer session does rush the topics a bit more than I would like.

**Reply by tempcooper** [reply]
> it would be to make it longer, so we have time to think about the problems. @Aidan-Myatt 

> could change one thing about this course, it would be the time I am taking it, as the summer session does rush the topics a bit more than I would like. @Aarons-Joshua 

We are going to be rushing a bit, but hopefully the time we spend is high-quality+high-density learning. _You got this :+1: !!_

**Comment by RyanBottass** [comment]
Hello, My name is Ryan, I am a mechanical engineering student at UConn going into my senior year. An interesting fact about me is that I have trained my cat, George, to sit on command (most of the time).  I am taking this course over the summer to try to lighten the load of classes for my senior year at UConn. I think the setup of the course is logical and I appreciate the use of google calendars as they are definitely useful in staying organized, especially in a faster pace summer course.

**Reply by cooperrc** [reply]
> An interesting fact about me is that I have trained my cat, George, to sit on command (most of the time)

:rofl: Have you noticed a statistically signifcant relation between command and action?

> google calendars as they are definitely useful in staying organized, especially in a faster pace summer course

I agree, That's why I migrated us to Classroom. Every assignment autoloads into the calendar so you can manage it from any device/program that handles .ical events.

**Comment by feliciones** [comment]
Hello everyone, my name is Simon Felicione and I am an incoming Senior Mechanical Engineering student. One interesting fact about me is that I did a brief apprenticeship where I began learning how to glass blow. I am here in order to lighten my course load over the year, and I am excited to learn more python and test my skills. I wish this course would not have to be so rushed and it would be a little more spread out in order to learn a bit more material in greater depth.

**Reply by cooperrc** [reply]
> brief apprenticeship where I began learning how to glass blow

My brother did that for ~6 years in Stonington. Where did you practice?

> wish this course would not have to be so rushed

It's not _too_ bad if you think of it as a 5-week full-time course. It gets difficult to juggle the course+job+family though, but **you got this :+1:**

**Reply by feliciones** [reply]
I actually did it in Fairfield, CT at Hotspot Glass Studio. I did it as a senior independent study in high school.

**Comment by y1bo44** [comment]
Hello everyone,
My name is Yibo Lin.One interesting fact about me is that I am now making videos on a platform called Bilibili.Its a foreign and very cool video platform.I made gaming videos and already received 10k followers.
I am here to learn more skills on the road of being an engineer especially on coding stuff.I am both excited and a little bit worried cause I do not have lots of experience with python.Hope I can try my best and getting better by professor's guidence.

**Reply by cooperrc** [reply]
> now making videos on a platform called Bilibili.Its a foreign and very cool video platform.I made gaming videos and already received 10k followers.

Whoah, that's great!

> I am both excited and a little bit worried cause I do not have lots of experience with python

Its good to be a _little_ worried. It gives you a little extra push to not fall behind. Make sure you're sitting down at least once/day to either find one answer _or_ one question. Either way that is _great_ progress. Many times, as I type out my question in a forum chat or github issue, I learn the solution as I review the documentation or error codes. Even if I don't, someone else will give you an answer and you're both better off from the the interaction. __Stay engaged, Stay positive!__

**Comment by caroline-cook** [comment]
Hi everyone! My name is Caroline Cook. An interesting fact about me is that I have gone skydiving before. 
I am here because I am a mechanical engineering major and I am hoping to take advantage of the summer by progressing with my coursework. I really liked the layout of Dynamics with Prof. Cooper last summer, so I knew this would be an interesting class to also enhance my knowledge in things like python and GitHub. I don't think there are any changes I would like to make to the course at this time. I really like interactivity and creativity aspect of the course, so I look forward to getting to edit the Jupyter notebooks as I learn through the material.

**Reply by tempcooper** [reply]
Welcome to another Cooper course, @caroline-cook ! Thanks for coming back.

> An interesting fact about me is that I have gone skydiving before

:cloud: :diving_mask: Very cool! I have not tried it yet, but I do know how to calculate terminal velocity during a dive.

**Comment by JordanAlnaimi** [comment]
Hi everyone, My name is Mohammed Alnaimi, Interesting fact about me is that I own a motorcycle and I enjoying riding, I feel like I clear my Brian every time I go for a ride and I enjoy working on it and improve the way its function. I am mechanical engineering major and I want to take this class, first, its required for my major and I believe that it will help me learn about coding since I think its very important for me to know for now and after graduation. I understand that time for summer classes is very short, if I can I would make it little bit longer I feel like I am in challenge to learn in this short time, I did take a summer class with professor Cooper in the past and it was fun and effective in short term so, I am really excited and I hope I make it to the end.

**Reply by cooperrc** [reply]
> Interesting fact about me is that I own a motorcycle and I enjoying riding

Nice :motorcycle: I'm a cyclist myself, but I agree there is something Zen-like when you're on two wheels in the open air. 

> I did take a summer class with professor Cooper

Welcome back, thanks for joining us again!

**Comment by VictoireHA** [comment]
Hello, my name is Huanran and I go by Mars. An interesting fact about me is that I speak 3 languages. I am a mechanical engineering major, and I am here to get ahead and reduce the workload for the next semester. The course provides an interesting aspect of learning python, because it is so popular now. Also it integrates python and traditional mechanics. I would add a few more optional tasks/assignments that elaborate on programming. It is always good to have more time to learn about the course.

**Reply by tempcooper** [reply]
> An interesting fact about me is that I speak 3 languages

Tres bon!

> I would add a few more optional tasks/assignments that elaborate on programming.

The "optional" assignments would be to do all of the exercises and examples in the reading notebooks e.g. [M01-notebook_02 exercise](https://cooperrc.github.io/computational-mechanics/module_01/02_Working_with_Python.html#id4)

I won't grade the notes/exercise in the notebooks and you don't have to turn them in, but there are ton of examples that you can change/add-to/refactor/etc. in the readings.

**Comment by yams11** [comment]
My name is Hannah Hyams. I am a Biomedical and Mechanical Engineering student on the premed track. I am from Kearny, N.J. and an interesting fact about me is my favorite animal (if you consider fantastical beasts) is a dragon. I am here because I it is a course requirement but I believe it will strengthen my abilities to apply mathematical and engineering principles to computational techniques.

**Reply by cooperrc** [reply]
:dragon: nice!

> will strengthen my abilities to apply mathematical and engineering principles to computational techniques

absolutely! this is a great reason to join the class. When I first started engineering courses, it seemed like everything was a math equation, but then when I asked about friction, drag, fracture, all these natural phenomenon that you need to consider in a design, the answer seemed to be _you need a computer for that..._

One goal here: if you can create an equation (F=ma or conservation of energy or heat conduction etc.) you can find a way to get the computer to solve it for you.

**Comment by vwolf89** [comment]
My name is Valerie Wolf. I usually go by Val. I am a senior Mechanical Engineering student in Colorado. An interesting fact is I am leaving a 16 year career in Veterinary Medicine, most recently working as a Neurosurgery Department Anesthetist. 

I'm here to take this class so I can stay on track to graduate next May. This course will be helpful in performing computations quickly and accurately in a career in engineering.

I enjoy calculus (weird right?) and have already completed a Mathematics Minor. I find it takes me far longer to write code than it would to solve a lot of calculus equations by hand. I am hoping to flip-flop that fact and learn to use python/matlab in an efficient way so I can use it effectively in the field rather than just complete this class so I can move on.

**Comment by CaptDoctorPirateShip** [comment]
Howdy Y'all,

My name is Dan. I am a senior, a part-time student at UConn, and an interesting fact about myself is that I am currently working on publishing a children's book that I wrote and illustrated. 

 I am taking ME3255 this semester because I believe that the knowledge I receive throughout this course will help me in future courses such as senior design. This course will help my career because it will allow me to efficiently complete mechanics problems using advanced computational techniques. 

I do not know too much about the course because it is only week one, but I would like to improve my programming skills throughout the course, because the job market is demanding more technological literacy and programming skills.

**Comment by Kelvi03** [comment]
Hello everyone! My name is Kelvi Ziso, and I’m a Mechanical Engineering student at UConn. I’m interested in cars and hope to work in the automotive industry after graduating.

I’m taking this course to improve my programming and computational skills and learn how to apply them to engineering problems.

I’d like to see more hands-on examples and projects that connect what we learn to real-world engineering applications.

**Comment by Dj-Dima** [comment]
Hi! My name is Djibring Galadima, I usuall go by Dj, Im studying Mechanical Engineering at Uconn like most of you and im a senior. Im interested Hiking, Basketball, Anime and Power Generation. Fun fact about me is i lived on a submarine for about 4 years or so.
I'm here because, it is part of the mechanical Engineering corriculum and I want to really understand the mechanics behind the simulation software to further grow my as a mechanical engineer. 
I would love to see specific challenges faced by engineers on the field (more specifically in for energy and power generation/optimization if applicable) as long with the overall intro to the broad applications.

**Comment by ChristianD8228** [comment]
What's up?!    
My name is Christian Donatucci, an interesting fact about me is that I ride motorcycles - well, one at a time. it's hard to ride multiple at once. 
I'm here as part of the ME curriculum, however i do have an interest in being able to understand the coding world to help me build my own projects. I'm simultaneously intimated and amazed by the capabilities of code, i mean - look at what we have in present day tech - it's amazing. So, I'm just looking to start taking some baby steps towards a greater understanding of what the human mind can create. Something I would change about the course that could help me achieve my goals would be to tailor it to my interests, or at least focusing on common products or items that are popular nowadays. its a selfish approach but true nonetheless because i could gain some useful footing that could prove itself useful to larger, more complicated personal projects in the future!
